packages/horizontal-slider-vdom/src/HorizontalSlider.tsx

```diff
@@ -0,0 +1,163 @@
+/** @jsx createElement */
+import {
+  FrameworkProps,
+  HorizontalSliderProps,
+  HorizontalSliderTranslations,
+  RecordWithObjectID,
+  Renderer,
+} from './types';
+
+let lastHorizontalSliderId = 0;
+
+export function generateHorizontalSliderId() {
+  return `uic-horizontal-slider-${lastHorizontalSliderId++}`;
```

Comment:
does this generate stable ids for use in server side rendering?

Reply:
This is the same logic as we used in https://github.com/algolia/autocomplete but I didn't tried it yet with SSR!

Reply:
For this to be SSR-safe, we need to accept an `id` prop in all our components. You would need to generate the `id` on the server and pass it to the client. (Autocomplete allows this.)

We've planed to add this later. But as it is now, it's not SSR-safe.

(Other interesting resources could be the new React's [`useOpaqueIdentifier`](https://github.com/facebook/react/pull/17322).)